lib/pbench/agent/utils.py

```diff
@@ -140,11 +140,18 @@ def __init__(self, spec: str, def_host_name: str):
             self._repr = f"{self.name} - {self.host}:{self.port}"
 
         self.pid_file = None
+        self.pid = None
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return self._repr
 
-    def kill(self, ret_val: int):
+    def get_pid(self) -> int:
+        if not self.pid:
+            raw_pid = self.pid_file.read_text()
+            self.pid = int(raw_pid)
+        return self.pid
```

Comment:
Since there's only one caller (and a single call?) to this function, is the caching of the return value really worthwhile?  (I like the encapsulation, though.)

Reply:
There are actually two places we call `kill'; though the caching arguably *is* a bit excessive as they don't occur in the same process. It just seemed like the right thing to do. 🍤